unittests/test_pipeline.py

```diff
@@ -287,6 +287,12 @@ def test_supports_system(self):
         assert not test.supports_system('testsys:gpu')
         assert not test.supports_system('testsys:login')
 
+        test.valid_systems = ['*:gpu']
+        assert test.supports_system('testsys:gpu')
+        assert test.supports_system('foo:gpu')
+        assert not test.supports_system('testsys:cpu')
+        assert not test.supports_system('testsys:login')
```

Comment:
You could add some additional assertions for testing the `*:*` and `system:*` alternatives.